### DISCUSSION TITLE
Connect click event to mark and emit event

### DISCUSSION THREAD

**Discussion by cozybuild** [ueberdosis/tiptap#1375] [General Questions]
I've got a new mark with some attributes and want to trigger an event when the mark is clicked. The editor is mounted in a Vue.js file - is it possible to trigger up an event from a mark throught the editor to the Vue.js component?

```
const MyExtension = Mark.create({
  name: 'myextension',
  ...
  ...
  addAttributes() {
    return {
      custom: {
        renderHTML: attributes => {
          return {
            'data-custom': attributes.custom,
            onclick: `console.log(this, ${attributes.custom})` // Here the event should be emitted to the editor / Vue.js component
          }
	}
      }
    }
  }
})
```

Currently I'm binding a click event to the markup and could include a store which redirects data to the component but maybe there is an easier solution for that?

**Comment by sipec** [comment]
in the `Link` component they do this via a prosemirror plugin

**Reply by sipec** [reply]
This approach didn't work for me because the plugin uses an onclick handler over the entire editor. Instead, for my use case in `renderHTML` I just created the html element the normal js way in order to add a click handler

```javascript
const MyExtension = Mark.create({
  name: 'myextension'
  ...
  renderHTML({ HTMLAttributes }) {
    const elem = document.createElement('span')

    Object.entries(
      mergeAttributes(this.options.HTMLAttributes, HTMLAttributes)
    ).forEach(([attr, val]) => elem.setAttribute(attr, val))

    elem.addEventListener('click', () => {
      elem.setAttribute('class', 'something-amazing')
    })

    return elem
  },
}
```
